global/scripts/MyelinMap_BC.sh

```diff
@@ -0,0 +1,168 @@
+    export HCPPIPEDIR="$(dirname -- "$0")/../.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+source "$HCPPIPEDIR/global/scripts/tempfiles.shlib"
+
+defaultSigma=$(echo "sqrt(200)" | bc -l)
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "corrects an individual native mesh myelin map for bias fields based on a group average myelin map"
+
+#mandatory
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject' 'Subject' '100206' "one subject ID"
+opts_AddMandatory '--registration-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files, e.g. 'MSMAll' or 'MSMSulc'"
+opts_AddMandatory '--msm-all-templates' 'MSMAllTemplates' 'path' "path to directory containing MSM All template files, e.g. 'YourFolder/global/templates/MSMAll'"
+#optional inputs
+opts_AddOptional '--use-ind-mean' 'UseIndMean' 'YES or NO' "whether to use the mean of the subject's myelin map as reference map's myelin map mean , defaults to 'YES'" 'YES'
+opts_AddOptional '--mcsigma' 'CorrectionSigma' 'number' "myelin map bias correction sigma, default '$defaultSigma'" "$defaultSigma"
+opts_AddOptional '--myelin-target-file' 'MyelinTarget' 'string' "alternate myelin map target, relative to the --msm-all-templates folder" 'Q1-Q6_RelatedParcellation210.MyelinMap_BC_MSMAll_2_d41_WRN_DeDrift.32k_fs_LR.dscalar.nii'
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+# Verify required environment variables are set and log value
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
+
+#display the parsed/default values
+opts_ShowValues
+
+# ------------------------------------------------------------------------------
+#  Main processing of script.
+# ------------------------------------------------------------------------------
+log_Msg "Starting main functionality"
+Caret7_Command=${CARET7DIR}/wb_command
+UseIndMeanBool=$(opts_StringToBool "$UseIndMean")
+# default folders
+SubjFolder=${StudyFolder}/${Subject}
+log_Msg "SubjFolder: $SubjFolder"
+AtlasSpaceFolder=${SubjFolder}/MNINonLinear
+log_Msg "AtlasSpaceFolder: $AtlasSpaceFolder"
+T1wFolder=${SubjFolder}/T1w
+log_Msg "T1wFolder: $T1wFolder"
+NativeFolder=${AtlasSpaceFolder}/Native
+log_Msg "NativeFolder: $NativeFolder"
+NativeT1wFolder=${T1wFolder}/Native
+log_Msg "NativeT1wFolder: $NativeT1wFolder"
+# MSMAll templates Myelin Target which is the group average MyelinMap_BC
+MyelinTarget="${MSMAllTemplates}/${MyelinTarget}"
+# 32k setting
+LowResMesh=32
```

Comment:
I'm not sure if we want to hardcode this, or have it the default with an option to change it.  It looks like the downsampled version of the uncorrected map is produced in the subject folder, and not deleted, so this would show up if we ever changed the default fMRI surface resolution.

Reply:
Agree this should be a parameter.